Fix navbar and footer overlap, improve footer button layout and image

PAINj9 opened the pull request:
- Made navbar and footer fixed without overlapping content
- Improved layout and spacing of Sign Up / Login buttons in the logged-out footer
- Added better spacing between images inside each CompetitionCard

PAINj9 commented:
I need to fix it yet, I didn't check that when you press a button, everything is under the footer
![Screenshot_2](https://github.com/user-attachments/assets/0037633a-0239-422d-b25a-0f6b5377767f)

siderdk commented:
Good Job Juani
If you could also add the dropdown (i guess it should be called pop up menu) with the logout button on the mobile navbar, also with the link to the profile page. it would be really awesom

PAINj9 commented:
> Good Job Juani If you could also add the dropdown (i guess it should be called pop up menu) with the logout button on the mobile navbar, also with the link to the profile page. it would be really awesom

I will add the menu on the mobile navbar with the link to profile page and logout button now

PAINj9 commented:
- Added hamburger menu for mobile navbar view
- Implemented working profile and logout links in the mobile dropdown
- Fixed footer profile icon to redirect to user profile page
- Moved notifications bell to avoid overlap with hamburger menu
- Removed unwanted horizontal scroll on /profile page
- Adjusted layout height handling to prevent overflow issues

siderdk commented:
here is what I'm getting : 
![image](https://github.com/user-attachments/assets/6fd0e841-4249-466e-a6a2-39dd72078972)

top bar does not have a hamburger menu. and I think in terms of UX, on mobile view, maybe it's better that menu (profile, and logout) is visible when clicking the "user icon" in the footer instead. 

Have you checked the company navbar?

te-online commented:
@siderdk Is your review a rejection or are you two still working on this or is it approved? 😅 😊

siderdk commented:
> @siderdk Is your review a rejection or are you two still working on this or is it approved? 😅 😊

The PR as it stands, does not do what it claims to. so it still is work in progress.

siderdk commented:
I will close this one so it's not confusing

siderdk commented:
You can always reopen it if you need to.
